### DISCUSSION TITLE
[Q] images show incorrectly when insert multiple images in same Excel's cell

### DISCUSSION THREAD

**Discussion by thavornprasertkaew** [exceljs/exceljs#2359] [Q&A]
Please help me 

      tl: { col: 6, row: 4.5 }  "row: 4.5" not calculator default height of row in exceljs  image is small not half

 worksheet.getRow(4).height = 200
            worksheet.addImage(logo, {
              tl: { col: 6, row: 4 },
              br: { col: 7, row: 4.5 },
              ext: { width: 100, height: 100 },
              editAs: 'absolute'
            })
![image](https://github.com/exceljs/exceljs/assets/82131638/b2ebf4ef-becf-4818-b142-03647e068b95)

**Comment by zurmokeeper** [comment]
@thavornprasertkaew Can you describe your problem a little more? I'm not reading what you're trying to ask. Also post the full code and upload the excel file?

**Comment by thavornprasertkaew** [comment]
import { Injectable } from '@angular/core';
import * as Excel from "exceljs";
import { saveAs } from 'file-saver';
import * as moment from 'moment';
import { HttpClient } from '@angular/common/http';


@Injectable({
  providedIn: 'root'
})
export class ExportExcelFollowlistService {
  constructor(private http: HttpClient) { }
  data: any
  logo: any
  public async exportExcel(excelData: any) {
    const title = excelData.followup_id;
    const startDate = excelData.date_assignment;
    const project_name = excelData.project_name;
    const header = excelData.header; // Update this with your actual header data
    const data = excelData.data;


    // Create a workbook with a worksheet
    let workbook = new Excel.Workbook();
    const worksheet = workbook.addWorksheet('Follow up List', {
      views: [
        { zoomScale: 75 }
      ]
    });

    // Header Rows
    worksheet.addRow([title]);
    worksheet.addRow([project_name]);
    worksheet.addRow([startDate]);

    // Apply formatting to the header rows
    const headerRows = [1, 2, 3];
    headerRows.forEach((rowNumber) => {
      let row = worksheet.getRow(rowNumber);
      row.height = 42;
      row.eachCell((cell, number) => {
        cell.fill = {
          type: 'pattern',
          pattern: 'solid',
          fgColor: { argb: 'FFFFFF' },
          bgColor: { argb: '' }

        };
        cell.border = {
          top: { style: 'thin' },
          left: { style: 'thin' },
          bottom: { style: 'thin' },
          right: { style: 'thin' }
        };
        cell.alignment = { vertical: 'middle', horizontal: 'center' };
        cell.font = {
          name: 'TH SarabunPSK',
          bold: true,
          color: { argb: '000000' },
          size: 30
        };
      });
    });

    // Header Row (main header)
    let headerRow = worksheet.addRow(header);

    headerRow.eachCell((cell, number) => {
      cell.fill = {
        type: 'pattern',
        pattern: 'solid',
        fgColor: { argb: '8A2BE2' },
        bgColor: { argb: '8A2BE2' },

      };

      cell.border = {
        top: { style: 'thin' },
        left: { style: 'thin' },
        bottom: { style: 'thin' },
        right: { style: 'thin' }
      };
      cell.alignment = { vertical: 'middle', horizontal: 'center' };
      cell.font = {
        name: 'TH SarabunPSK',
        bold: true,
        color: { argb: 'FFFFFF' },
        size: 14
      };
    });

    // ...
    for (const [index, rowData] of data.entries()) {

      if (rowData[14] && Array.isArray(rowData[14])) {
        const imageSize = { width: 200, height: 200 };
        const dataImage = rowData[14]
        for (let i = 0; i < dataImage.length; i++) {
          const image = await this.getImageData(dataImage[i]);
          const logo = workbook.addImage({
            base64: image,
            extension: 'png',
          });


          const rowitem = 4
          if (i === 0) {
            const imageCellPosition = { col: 14, row: rowitem + index };
            worksheet.addImage(logo, {
              tl: { col: imageCellPosition.col, row: imageCellPosition.row },
              br: {col: imageCellPosition.col + 1, row: imageCellPosition.row + 0.5 },
              ext: { width: 100, height: 100 },
              editAs: 'absolute'
            })
          }
          else {

            const imageCellPosition = { col: 14, row: rowitem + index };
            worksheet.addImage(logo, {
              tl: { col: imageCellPosition.col, row: imageCellPosition.row + 0.5 },
              br: {col: imageCellPosition.col + 1, row: imageCellPosition.row + 1 },
              ext: { width: 100, height: 100 },
              editAs: 'absolute'
            })
          }
        }
        rowData[14] = ''

      }


      this.data = worksheet.addRow(rowData)
        this.data.height = 200
     
       const cellIndices = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16];
       cellIndices.forEach((index) => {
         let cell = this.data.getCell(index);
 
         cell.border = {
           top: { style: 'thin' },
           left: { style: 'thin' },
           bottom: { style: 'thin' },
           right: { style: 'thin' }
         };
         cell.font = {
           name: 'TH SarabunPSK',
           bold: true,
           color: { argb: '000000' },
           size: 14
         };
         cell.alignment = { vertical: 'middle', horizontal: 'left' };
         cell.alignment.wrapText = true; 
       }); 
    }


    // ...


    worksheet.getColumn(1).width = 6.71;
    worksheet.getColumn(2).width = 23.86;
    worksheet.getColumn(3).width = 23.86;
    worksheet.getColumn(4).width = 42.57;
    worksheet.getColumn(5).width = 16.71;
    worksheet.getColumn(6).width = 26.43;
    worksheet.getColumn(7).width = 21.86;

    worksheet.getColumn(8).width = 26.29;
    worksheet.getColumn(9).width = 19.29;
    worksheet.getColumn(10).width = 16.86;
    worksheet.getColumn(11).width = 12.43;
    worksheet.getColumn(12).width = 10.00;
    worksheet.getColumn(13).width = 17.29;
    worksheet.getColumn(14).width = 32.00;
    worksheet.getColumn(15).width = 32.00;
    worksheet.getColumn(16).width = 32.00;

    /*  // Adjust column widths
     worksheet.columns.forEach((column) => {
       // Adjust width based on your preference
       column.width = 12;
     }); */

    // Merge title, project_name, and date range cells
    worksheet.mergeCells(`A1:P1`);
    worksheet.mergeCells(`A2:P2`);
    worksheet.mergeCells(`A3:P3`);


  
    // Generate & Save Excel File
    workbook.xlsx.writeBuffer().then((data) => {

      const fileName = `${title}.xlsx`;
      let blob = new Blob([data], { type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet' });

      saveAs(blob, fileName);
    });

  }

  private async getImageData(imageUrl: string): Promise<string> {
    const response = await this.http.get(imageUrl, { responseType: 'arraybuffer' }).toPromise();
    const imageBase64 = btoa(
      new Uint8Array(response)
        .reduce((data, byte) => data + String.fromCharCode(byte), '')
    );
    return `data:image/png;base64,${imageBase64}`;
  }
}

**Comment by thavornprasertkaew** [comment]
### I want the images to be in the same Row but the images come out very small. 


if (i === 0) {
        const imageCellPosition = { col: 14, row: rowitem + index };
        worksheet.addImage(logo, {
          tl: { col: imageCellPosition.col, row: imageCellPosition.row },
          br: {col: imageCellPosition.col + 1, row: imageCellPosition.row + 0.5 },
          ext: { width: 100, height: 100 },
          editAs: 'absolute'
        })
      }
      else {

        const imageCellPosition = { col: 14, row: rowitem + index };
        worksheet.addImage(logo, {
          tl: { col: imageCellPosition.col, row: imageCellPosition.row + 0.5 },
          br: {col: imageCellPosition.col + 1, row: imageCellPosition.row + 1 },
          ext: { width: 100, height: 100 },
          editAs: 'absolute'
        })
      }
    }

**Comment by z3151zaq** [comment]
I encountered this problem, too. 

![image](https://github.com/exceljs/exceljs/assets/28192094/852d718e-11cc-45d7-951c-6d563bd36c16)

Here is my code. I tried to put two images in one cell. They are not supposed to overlap. It seems 0.5 are not calculated as half of the width of the cell but the __default cell width of the worksheet__.

```JS
i.imgIds.forEach((id: number, idx: number) => {
      ws.addImage(id, {
        tl: {
          col: headers.length - 2 + 0.5 * idx,
          row: extra.length + index + 1,
          colWidth: 85,
        },
        ext: { width: 120, height: 120 },
      });
    });
```
